apps/condo/domains/resident/tasks/notifyResidentsOnPayday.js

```diff
@@ -100,32 +103,56 @@ async function notifyResidentsOnPayday () {
 
         for (const consumer of consumers) {
             try {
-                const userId = consumer.resident.user.id
-                const organizationId = consumer.organization.id
-
-                //checking whether this user has been processed
-                if (state.recipientsMap.has(organizationId) && state.recipientsMap.get(organizationId) === userId) continue
-                state.recipientsMap.set(organizationId, userId)
+                const accountNumber = get(consumer, ['accountNumber'])
 
                 const receipts = await loadListByChunks({
                     context,
                     chunkSize:20,
                     list: BillingReceipt,
                     where: {
+                        account: {
+                            number: accountNumber,
+                        },
                         context: {
                             id: consumer.billingIntegrationContext.id,
                         },
-                        period: consumer.billingIntegrationContext.lastReport.period,
+                        period_gte: dayjs().subtract(2, 'month').startOf('month').toISOString(),
```

Comment:
Why period filter is become relative to the current month instead of lastReport.period?

Reply:
because receipts can be uploaded in any order. If you first download a receipt for November, then for September, the September receipt will be considered the last, although in fact the November one has already been downloaded and is current